### PR TITLE
Clean up the pending task

### LINKED ISSUE #696: run_sync does not clean up the pending task after run_until_complete is interrupted
When the wrapped method (like `BlockingKernelClient.get_shell_msg`) is interrupted by KeyboardInterrupt(Ctrl+C), the event loop is interrupted, and the task ( `asyncio.Task`  ) becomes a pending state.
If I call the wrapped method again, it does not return.

I think `run_sync` should cancel the pending task.
https://stackoverflow.com/questions/30765606/whats-the-correct-way-to-clean-up-after-an-interrupted-event-loop

### PR DESCRIPTION
**Pull request by shingo78**
Fix #696

### CODE REVIEW

File: jupyter_client/utils.py [hunk 1]
```diff
@@ -18,7 +18,12 @@ def wrapped(*args, **kwargs):
         import nest_asyncio  # type: ignore
 
         nest_asyncio.apply(loop)
-        return loop.run_until_complete(coro(*args, **kwargs))
+        future = asyncio.ensure_future(coro(*args, **kwargs))
+        try:
+            return loop.run_until_complete(future)
+        except KeyboardInterrupt:
+            future.cancel()
+            raise
```
Comment (davidbrochart) [inline jupyter_client/utils.py:26]: Maybe we want to handle any kind of exception?

```suggestion
        except BaseException as e:
            future.cancel()
            raise e
```

**Review by davidbrochart — COMMENTED** [review]
Thanks, that looks good. It would be great to add a test.

File: jupyter_client/utils.py (hunk 1 above)
Comment (shingo78) [reply jupyter_client/utils.py:26]: I agree with your suggestion.

**Comment by davidbrochart** [comment]
Great, do you feel like adding a test?

**Comment by shingo78** [comment]
Yes, I would like to add a test.

**Comment by shingo78** [comment]
Please review the test I pushed.

**Review by davidbrochart — APPROVED** [review]

**Comment by davidbrochart** [comment]
Thanks @shingo78, that looks great.

**Pull request merged**